ingenialink/configuration_file.py

```diff
@@ -0,0 +1,458 @@
+        if major_version != cls._SUPPORTED_MAJOR_VERSION:
+            raise NotImplementedError(
+                f"Configuration file not supported: "
+                f"Supported version: {cls._SUPPORTED_MAJOR_VERSION}, "
+                f"File version: {major_version}"
+            )
+        body = cls._find_and_check(root, cls.__BODY_ELEMENT)
+        device_element = cls._find_and_check(body, Device.ELEMENT_NAME)
+        registers_element = cls._find_and_check(device_element, cls.__REGISTERS_ELEMENT)
+        register_element_list = cls._findall_and_check(
+            registers_element, ConfigRegister.ELEMENT_NAME
+        )
+        config_device = Device.from_xcf(device_element)
+        xcf_instance = cls(config_device)
+        for reg in register_element_list:
+            try:
+                xcf_instance.add_config_register(ConfigRegister.from_xcf(reg))
+            except (ValueError, KeyError) as e:  # noqa: PERF203
+                logger.warning(f"{reg}: {e}")
+        xcf_instance.major_version = major_version
+        xcf_instance.minor_version = minor_version
+        return xcf_instance
+
+    @classmethod
+    def create_xcf(
+        cls,
+        interface: Interface,
+        part_number: Optional[str],
+        product_code: Optional[int],
+        revision_number: Optional[int],
+        firmware_version: Optional[str],
+        node_id: Optional[int] = None,
+    ) -> "ConfigurationFile":
+        """Create an empty XCF with the device info.
+
+        Args:
+            interface: drive interface
+            part_number: drive part number
+            product_code: drive product code
+            revision_number: drive firmware revision number
+            firmware_version: drive firmware version
+            node_id: drive node id. Only for CANopen drives
+
+        Returns:
+            creates a XCF instance
+        """
+        device = Device(
+            interface, part_number, product_code, revision_number, firmware_version, node_id
+        )
+        xcf_instance = cls(device)
+        return xcf_instance
+
+    def add_register(self, register: Register, value: Union[float, int, str, bool]) -> None:
+        """Add register to the XCF class.
+
+        Args:
+            register: register that will be added to the XCF
+            value: value that will be added to the register in te XCF
+
+        """
+        config_register = ConfigRegister.from_register(register, value)
+        self.add_config_register(config_register)
+
+    def add_config_register(self, config_register: ConfigRegister) -> None:
+        """Add ConfigRegister to the XCF class.
+
+        Args:
+            config_register: register that will be added to the XCF
+        """
+        self.__registers.append(config_register)
+        self.__subnodes.add(config_register.subnode)
+
+    def to_xcf(self, xcf_path: str) -> None:
+        """Save a file with the config file in the target path.
```

Comment:
ConfigFile.create_xcf() and to_xcf() will have to be created on the same procedure, naming wise it's a bit confusing

ConfigFile.create(), config_file.save_to_xcf() ??

Reply:
You are right, it's confusing. I think the main problem is the create_xcf function, I suggest create_empty_configuration(). About to_xcf, I like it because exist another function named from_xcf, but I can change both to: load_from_xml() and save_to_xml(). I think this is totally clear.